scipy/signal/tests/test_signaltools.py

```diff
@@ -129,6 +128,14 @@ def test_same_mode(self):
         d = array([57, 61, 63, 57, 45, 36])
         assert_array_equal(c, d)
 
+    def test_convolve_method(self):
+        np.random.seed(42)
+        x = np.random.rand(100)
+        h = np.random.rand(50)
+
+        assert_allclose(convolve(x, h, method='direct'), 
+                        convolve(x, h, method='fft'))
+
```

Comment:
and test error messages like 

    assert_raises(ValueError, convolve, Decimal(3), Decimal(4), method='fft')

I think `_TestCorrelateReal` already tests that Decimal object arrays work correctly even with `method='auto'`, because it's the default?

but I don't think `test_method` should be inside of `_TestCorrelateReal`?  Because `_TestCorrelateReal` gets converted into multiple classes, one for every data type, but you're not using that data, so `test_method` will be called repeatedly for no reason?